src/common/transformations/include/transformations/op_conversions/group_query_attention_decomposition.hpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/op/group_query_attention.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/pass/matcher_pass.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+namespace pass {
+
+class TRANSFORMATIONS_API GroupQueryAttentionDecomposition;
+
+}  // namespace pass
+}  // namespace ov
+
+class ov::pass::GroupQueryAttentionDecomposition : public ov::pass::MatcherPass {
+public:
+    OPENVINO_MATCHER_PASS_RTTI("GroupQueryAttentionDecomposition");
+    GroupQueryAttentionDecomposition();
+
+private:
+    ov::OutputVector decompose(std::shared_ptr<ov::op::GroupQueryAttention> node);
```

Comment:
minor:
as I can see, these functions don't use any members of GroupQueryAttentionDecomposition and can be used separately,
so it's better to move it to unnamed namespace inside group_query_attention_decomposition.cpp file or to utils if we plan to re-use it